File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/RsaGenerateRequest.java
package org.cloudfoundry.credhub.requests;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

public class RsaGenerateRequest extends BaseCredentialGenerateRequest {
  @JsonProperty("parameters")
  private RsaGenerationParameters generationParameters;

  @Override
  @JsonIgnore
  public GenerationParameters getGenerationParameters() {
    if (generationParameters == null) {
      generationParameters = new RsaGenerationParameters();
    }
    return generationParameters;
  }

  public void setGenerationParameters(final RsaGenerationParameters generationParameters) {
    this.generationParameters = generationParameters;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/RsaGenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

class RsaGenerateRequest : BaseCredentialGenerateRequest() {
    @JsonProperty("parameters")
    override var generationParameters: RsaGenerationParameters? = null
        @JsonIgnore
        get() {
            if (field == null) {
                field = RsaGenerationParameters()
            }
            return field
        }
}
